### ISSUE TITLE
Rename `docs` module `spring-cloud-gcp-docs` back

### ISSUE THREAD

**Issue by artembilan** [spring-cloud/spring-cloud-gcp#554]
Since this project is not part of the Spring Cloud train yet, and we need to publish Reference Manual on the project page anyway, the current proposal to fix the problem is to revert the change via #330 and come back to the subject of the Train when everybody is ready for that, include specific changes to the build process to publish Docs and JavaDocs properly.

/CC @dsyer

**Comment by joaoandremartins** [comment]
I'll do this one after #546 is merged

**Comment by joaoandremartins** [comment]
It would also pretty cool to get the refdoc with the latest layout (i.e., the one that gets generated in the `generated-docs` folder)...
